gpbft/gpbft.go

```diff
@@ -797,10 +807,23 @@ func (i *instance) skipToRound(round uint64, chain ECChain, justification *Justi
 // Returns whether a chain is acceptable as a proposal for this instance to vote for.
 // This is "EC Compatible" in the pseudocode.
 func (i *instance) isCandidate(c ECChain) bool {
-	for _, candidate := range i.candidates {
-		if c.Eq(candidate) {
-			return true
-		}
+	_, exists := i.candidates[c.Key()]
+	return exists
+}
+
+func (i *instance) addCandidatePrefixes(c ECChain) bool {
+	var addedAny bool
+	for l := range c {
+		addedAny = i.addCandidate(c.Prefix(l)) || addedAny
+	}
```

Comment:
nit: I'd do this the other way and stop when we find the first prefix. I.e.:

```go
for i := len(c)-1; i > 0 && !addedAny; i-- {
    addedAny ||= i.addCandidate(c.Prefix(l))
}
```

Reply:
(not sure if `||=` is a thing...)

Reply:
Not in golang